extensions/sql-database-projects/src/models/project.ts

```diff
@@ -87,6 +88,30 @@ export class Project {
 		await this.updateImportedTargetsToProjFile(constants.NetCoreCondition, constants.NetCoreTargets, undefined);
 	}
 
+	private async updateAfterCleanTargetInProjFile(): Promise<void> {
+		// Search if clean target already present, update it
+		for (let i = 0; i < this.projFileXmlDoc.documentElement.getElementsByTagName(constants.Target).length; i++) {
+			const afterCleanNode = this.projFileXmlDoc.documentElement.getElementsByTagName(constants.Target)[i];
+			const name = afterCleanNode.getAttribute(constants.Name);
+			if (name === constants.AfterCleanTarget) {
+				return await this.createCleanFileNode(afterCleanNode);
```

Comment:
is this adding another Clean file node even if one already gets found?

Reply:
no no, if its file AfterClean target - it adds a child to it to add the clean file node. if it does nt find an after clean target - then it adds the target and then adds the child. Actually just writing a test on that :) will send shortly.

Reply:
ah ok I was initially confused because I didn't realize this was only called the one time the project gets updated for round tripping from SSDT